### PR TITLE
Match project pref and project correctly

### LINKED ISSUE #4263: Email project preferences aren't updated correctly
## Expected behavior
stop/start receiving emails when checkbox is un/ticked

## Current behavior
preferences are updated for the wrong project, because the order of projects and project preferences aren't matching.

## Steps to replicate
https://www.zooniverse.org/talk/17/507628

### PR DESCRIPTION
**Pull request by simoneduca**
Staging branch URL: https://email-proj-prefs.pfe-preview.zooniverse.org/

Fixes #4263 .

Describe your changes.
It displays project names keyed with `projectPref.links.project`

# Required Manual Testing

- [ ] Does the non-logged in home page render correctly?
- [ ] Does the logged in home page render correctly?
- [ ] Does the projects page render correctly?
- [ ] Can you load project home pages?
- [ ] Can you load the classification page?
- [ ] Can you submit a classification?
- [ ] Does talk load correctly?
- [ ] Can you post a talk comment?

# Review Checklist

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [ ] Are the tests passing locally and on Travis?

# Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

### CODE REVIEW

**Comment by simoneduca** [comment]
This is a WIP because I'm not sure why the new list of projects includes those without display name.
screen shot 2018-01-05 at 16 41 34

File: app/pages/settings/email.jsx [hunk 1]
```diff
@@ -59,6 +59,8 @@ TalkPreferences.propTypes = {
 };
 
 function ProjectPreferences({ projects, projectPreferences, onChange }) {
+  const projectNames = [];
```
Comment (eatyourgreens) [inline app/pages/settings/email.jsx]: Is this meant to be a hash, rather than an array? `const projectNames = {};`

**Comment by simoneduca** [comment]
The project list is now filtered.

**Comment by simoneduca** [comment]
Thanks @eatyourgreens, should make more sense now.

**Comment by eatyourgreens** [comment]
The callback to `Array.filter()` should return true/false, not `pref`. Your code works because `pref` evaluates to `true` but you shouldn't rely on implicit conversion. What you should filter on is `pref => projectIDs.includes(pref.links.project)`.

**Comment by simoneduca** [comment]
Gotcha, sorry about that. I think it's sorted now, thanks.

**Pull request merged**